### ISSUE TITLE
Tensorflow serving predict call to keras pre trained model always give same result

### ISSUE THREAD

**Issue by amithbk12man** [tensorflow/serving#972]
I have exported two models one keras pre built Xception model and one my own model when I try to call predict call to them in tensorflow serving I always get the same result. Please help me what is the mistake am doing. In the below code    " print(result2)" aways give same result.

def main(_):
    print(FLAGS.image)
    host, port = FLAGS.server.split(':')
    channel = implementations.insecure_channel(host, int(port))
    stub = prediction_service_pb2.beta_create_PredictionService_stub(channel)

    # Change test image to RGB mode
    # my code added
    img = image.load_img(FLAGS.image, target_size=(299, 299))
    a = image.img_to_array(img)
    a = np.expand_dims(a, axis=0)
    image_data = preprocess_input(a)
    #print(image_data)
    from keras.applications.xception import Xception
    img_array = np.vstack(image_data)
    model1 = Xception(weights='imagenet', include_top=False)
    model1._make_predict_function()
    print('now callng')
    value = model1.predict(image_data, verbose=1)

    # Create PredictRequest ProtoBuf from image data
    request = predict_pb2.PredictRequest()
    request.model_spec.name = "Xception"
    request.model_spec.signature_name = "prediction"
    print('calling function')
    request.inputs["images"].CopyFrom(
        tf.contrib.util.make_tensor_proto(image_data, dtype="float32", shape=[1,299,299,3]))

    result = stub.Predict(request, 10.0)

    stub2 = prediction_service_pb2.beta_create_PredictionService_stub(channel)
    request2 = predict_pb2.PredictRequest()
    request2.model_spec.name = "Xception1a"
    request2.model_spec.signature_name = "prediction"

    #request2.inputs["images"].CopyFrom(
    #    tf.contrib.util.make_tensor_proto(tensor_util.MakeNdarray(result.outputs["prediction"]), dtype="float32",shape=[1,10,10,2048]))

    request2.inputs["images"].CopyFrom(
        tf.contrib.util.make_tensor_proto(value, dtype="float32",shape=[1,10,10,2048]))
    result2 = stub2.Predict(request2, 10.0)

    print(result2)

if __name__ == '__main__':
  tf.app.run()

**Comment by duplex143** [comment]
You generally get same probabilities if you forget preprocessing. Do you need to preprocess `result `?

**Comment by amithbk12man** [comment]
I found out the issue. In case of exception which is a pre trained model I was not explicitly loading the weights. Yes, preprocessing is required in the client call and I did so. Now, it works.
